apps/trading/client-pages/home/home.tsx

```diff
@@ -17,18 +21,37 @@ export const Home = () => {
   const { data, error, loading } = useDataProvider({
     dataProvider: marketsWithDataProvider,
   });
-  const { update } = useGlobalStore((store) => ({
+  const { update, marketId } = useGlobalStore((store) => ({
     update: store.update,
+    marketId: store.marketId,
   }));
-
+  const variables = useMemo(
+    () => ({
+      marketId: marketId || '',
+    }),
+    [marketId]
+  );
+  const { data: singleMarketData, loading: singleMarketLoading } =
+    useDataProvider<SingleMarketFieldsFragment, never>({
+      dataProvider: marketProvider,
+      variables,
+      skip: !marketId,
+    });
+  const marketExists = Boolean(singleMarketData);
+  const marketNotExists = !marketExists && !singleMarketLoading;
   const { pageTitle, updateTitle } = usePageTitleStore((store) => ({
     pageTitle: store.pageTitle,
     updateTitle: store.updateTitle,
   }));
 
   useEffect(() => {
-    if (data) {
-      const marketId = data[0]?.id;
+    if (marketExists) {
+      navigate(Links[Routes.MARKET](marketId as string), {
+        replace: true,
+      });
+      update({ shouldDisplayWelcomeDialog: false });
```

Comment:
suggestion (non blocking): For me the home page logic should be only for deciding which market to redirect to. Is there any way we can change this so that we don't need this new piece of global state, and the WelcomeDialog component itself can contain the logic for deciding when to show and what dialog content to show?

Reply:
I'm gonna reconsider whole solution if it looks too messy and try to make it all simpler

Reply:
So, I pushed some changes. I added also some cleaning up of market page: moved update of last visiting market to update method of market data provider, rather than taking just param from url which can be easily misformatted.
Please @mattrussell36 take a look in a free while.